Files/UserControls/LayoutModes/GridViewBrowser.xaml.cs

```diff
@@ -219,12 +234,23 @@ private async void CommitRename(TextBox textBox)
 
         private void EndRename(TextBox textBox)
         {
-            StackPanel parentPanel = textBox.Parent as StackPanel;
-            TextBlock textBlock = parentPanel.Children[0] as TextBlock;
-            textBox.Visibility = Visibility.Collapsed;
-            textBlock.Visibility = Visibility.Visible;
+            if (App.AppSettings.LayoutMode == 2)
+            {
+                Popup popup = (textBox.Parent) as Popup;
+                TextBlock textBlock =
+                    (popup.Parent as Grid).Children[1] as TextBlock;
```

Comment:
The same, line is not too long. Line should be moved on new line, if length > 100